strawberryfields/ops.py

```diff
@@ -1803,39 +1795,34 @@ class Gaussian(Preparation, Decomposition):
     ns = None
 
     def __init__(self, V, r=None, decomp=True, tol=1e-6):
-        # TODO NOTE: there is no contextual hbar value anymore, is the hbar actually necessary here?
-
-        #: float: value of :math:`\hbar` used in the definition of the :math:`\x` and :math:`\p` quadrature operators
-        # TODO can we just divide V by hbar/2 here and remove hbar from further expressions?
-        self.hbar = sf.hbar
-        self.ns = V.shape[0]//2
+        # internally we eliminate hbar from the covariance matrix V (or equivalently set hbar=2), but not from the means vector r
+        V = V / (sf.hbar / 2)
+        self.ns = V.shape[0] // 2
 
         if r is None:
-            # all zeroes
-            r = [0] * (2*self.ns)
+            r = np.zeros(2*self.ns)
         r = np.asarray(r)
 
         if len(r) != V.shape[0]:
             raise ValueError('Vector of means must have the same length as the covariance matrix.')
 
+        super().__init__([V, r])  # V is hbar-independent, r is not
+
         self.x_disp = r[:self.ns]
         self.p_disp = r[self.ns:]
 
         if decomp:
             th, self.S = williamson(V, tol=tol)
-            self.pure = np.abs(np.linalg.det(V) - (self.hbar/2)**(2*self.ns)) < tol
-            self.nbar = np.diag(th)[:self.ns]/self.hbar - 0.5
-
-        super().__init__([V, r])
-
-        self.decomp = decomp
+            self.pure = np.abs(np.linalg.det(V) - 1.0) < tol
+            self.nbar = 0.5 * (np.diag(th)[:self.ns] - 1.0)
 
+        self.decomp = decomp  #: bool: if False, use the backend API call instead of decomposition
         # FIXME merge() probably does not work for Gaussians if r is not zero?
```

Comment:
What is status of this `FIXME`? Will it impact decomposition refactoring?

Reply:
Currently, merging of decompositions assumes matrix multiplication, which is not the case with displacements (need to be added instead). This caused no issues previously because no compilation/optimization was being performed, but this issue might now arise.

Reply:
Surprisingly the merging of `ops.Gaussian` instances works, because its first parent class is `Preparation`, and `Preparation.merge()` just says that when two Preparations are merged, only the latter one matters. `Decomposition.merge()` is never called. I removed the FIXME and updated a test.

Reply:
Oh, wow haha